src/compiler/scala/tools/nsc/transform/Delambdafy.scala

```diff
@@ -95,14 +95,16 @@ abstract class Delambdafy extends Transform with TypingTransformers with ast.Tre
 
       // no need for adaptation when the implemented sam is of a specialized built-in function type
       val lambdaTarget = if (isSpecialized) target else createBoxingBridgeMethodIfNeeded(fun, target, functionalInterface, sam)
+      val isSerializable = samUserDefined == NoSymbol || samUserDefined.owner.isNonBottomSubClass(definitions.JavaSerializableClass)
+      val addScalaSerializableMarker = isSerializable && samUserDefined == NoSymbol
```

Comment:
`isSerializable` is redundant here, i think it would be clearer to just have `samUserDefined == NoSymbol`